src/api/providers/fetchers/modelCache.ts

```diff
@@ -145,13 +145,130 @@ export const getModels = async (options: GetModelsOptions): Promise<ModelRecord>
 	}
 }
 
+/**
+ * Force-refresh models from API, bypassing cache.
+ * Uses atomic writes so cache remains available during refresh.
+ *
+ * @param options - Provider options for fetching models
+ * @returns Fresh models from API
+ */
+export const refreshModels = async (options: GetModelsOptions): Promise<ModelRecord> => {
+	const { provider } = options
+
+	let models: ModelRecord
+
+	try {
+		// Force fresh API fetch - skip getModelsFromCache() check
+		switch (provider) {
+			case "openrouter":
```

Comment:
I'm a little nervous about having this list of providers here in a way that we forget to update when we add a new provider. Any ideas on how to prevent that from happening?